filemutex_flock.go

```diff
@@ -57,3 +58,15 @@ func (m *FileMutex) RUnlock() {
 	}
 	m.mu.RUnlock()
 }
+
+// Close does an Unlock() combined with closing and unlinking the associated
+// lock file. You should create a New() FileMutex for every Lock() attempt if
+// using Close().
+func (m *FileMutex) Close() {
+	if err := syscall.Flock(m.fd, syscall.LOCK_UN); err != nil {
+		panic(err)
+	}
+	syscall.Close(m.fd)
+	syscall.Unlink(m.path)
```

Comment:
Yes, the idea is that each process does New() -> Lock() -> Close(), and only of the processes will hold the lock at any one time. This works (I think, not strictly tested) if Close() unlinks. I don't see the use case of needing the lock file to remain on disk once Close()d.